### DISCUSSION TITLE
help with deployment

### DISCUSSION THREAD

**Discussion by ghost** [ckan/ckan#7307] [Q&A]
Hi, I have been following the deployment documentation to get my CKAN build live (or atleast dev live so others can test).   But each time I complete the installation I am unable to action step6 restart nginx as it conflicts,  googling found the issue is due to apach2 being on port 80.   Kill the process and restart - everything looks good. 

but I am unable to see my site. 

I have checked to see if my site appears on 127.0.0.1:8080 - but it doesnt this is the port for jetty9,  

I am not sure what do next - does anyone have any thoughts please.

**Comment by kowh-ai** [comment]
jetty9 shouldn't be being used for Solr. It shouldn't be running at all. Did you follow these https://docs.ckan.org/en/latest/maintaining/installing/install-from-source.html#installing-solr-manually instructions for installing Solr ?

**Comment by ghost** [comment]
we did the install from source based on 2.9,  where Jetty came as part of the installs. 

Am I best firing up a new machine and installing based on latest ?  or can I get away with removing Jetty and making some changes?

**Reply by kowh-ai** [reply]
If it were me I'd fire up a new machine and install as per the **latest** docs and use the the latest CKAN 2.9

![Screenshot 2023-01-05 at 3 04 27 pm](https://user-images.githubusercontent.com/54408245/210798386-2ae892e5-6d0f-4692-a926-b0e8282ffcda.png)

**Comment by ghost** [comment]
Cheers Brett,   I will give it a go.   

Appreciate the help. 
Jack

**Comment by ghost** [comment]
no joy - I have created a fresh machine, installed from source onto ubuntu 20.04,  I can see the site in localhost / 127.0.0.1:5000,  

followed the deployment of source install again from latest but I cant get to the full site.  
I can see 127.0.0.1:8080 is now apache,

**Comment by kowh-ai** [comment]
mmm...Apache _or_ a WSGI Web Server?

**Comment by ghost** [comment]
apache,   


ive since moved the port from 8080 to 9000 to see if that helps incase there is a cross over somewhere. 

![image](https://user-images.githubusercontent.com/115991393/210844261-e860ec8d-a45d-41fe-9102-a2c8f434d96b.png)
 
as you can see nothing shows on that port either - i've never done hosting so this is all new to me...

**Reply by kowh-ai** [reply]
I see the **uwsgi** process listening on port 9000. BTW: no need to run CKAN separately (port 5000) to running uwsgi...

**Reply by ghost** [reply]
I am properly confused now. 

so If I install CKAN from source and run it on localhost:5000 to make sure its up,  then stop the process. 

follow the instructions for deployment and everything should be running?

**Reply by kowh-ai** [reply]
- Only 2 servers are running, NGINX as Remote Proxy, WSGI Server as the Web Server
- Flow like this: User <--> NGINX port=80/443 <--> WSGI Server port=8080 ( the WSGI Server is the entry into the application ie:CKAN )
- You can run `ckan -c /etc/ckan/default/ckan.ini run` if you like but it’s not needed -  by default it will run on port 5000

**Reply by kowh-ai** [reply]
WSGI Server is a better option than using the Apache Server

**Reply by ghost** [reply]
fully understand all of that, but I keep getting the same issue which is when I restart nginx and supervisor after changing the IP to listen to - I cant see the CKAN site on the given IP.  

I must be missing something obvious but cant think what.

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/115991393/211002272-7e68b5cc-3a59-47e8-93df-cac51c1e6d5e.png)

**Comment by ghost** [comment]
Brett, thank you for your help so far.   the issue is from my firms side and not CKAN so please consider this discussion closed.  

thank you for your time, but I now have to rebuild in Docker not ubuntu source. 

Thanks again
